app/gui2/shared/ast/tree.ts

```diff
@@ -749,6 +749,59 @@ export interface MutableUnaryOprApp extends UnaryOprApp, MutableAst {
 }
 applyMixins(MutableUnaryOprApp, [MutableAst])
 
+interface AutoscopedIdentifierFields {
+  operator: NodeChild<SyncTokenId>
+  identifier: NodeChild<SyncTokenId>
+}
+export class AutoscopedIdentifier extends Ast {
+  declare fields: FixedMapView<AstFields & AutoscopedIdentifierFields>
+  constructor(module: Module, fields: FixedMapView<AstFields & AutoscopedIdentifierFields>) {
+    super(module, fields)
+  }
+
+  static tryParse(
+    source: string,
+    module?: MutableModule,
+  ): Owned<MutableAutoscopedIdentifier> | undefined {
+    const parsed = parse(source, module)
+    if (parsed instanceof MutableAutoscopedIdentifier) return parsed
+  }
+
+  static concrete(module: MutableModule, operator: NodeChild<Token>, identifier: NodeChild<Token>) {
+    const base = module.baseObject('AutoscopedIdentifier')
+    const fields = composeFieldData(base, {
+      operator,
+      identifier,
+    })
+    return asOwned(new MutableAutoscopedIdentifier(module, fields))
+  }
+
+  static new(module: MutableModule, operator: Token, identifier: Token) {
```

Comment:
The `new` methods for `Ast` types are the safe, easy to use interface for creating instances (as opposed to `concrete`, which is for exactly-representing any data provided by the parser). As such there are a few things I'd do differently here:
- Since only one operator would be correct, rather than accept an operator `Token` it can be constructed in the method.
- Arbitrary tokens can't safely be accepted for the `identifier`, see comment on `setIdentifier` below.
- The `module` should be an optional parameter at the end (constructing a `MutableModule.Transient()` if omitted)--currently types are not consistent about this, but I have been moving toward this more convenient API.